cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -106,6 +106,7 @@ if (NOT CONFIG_NEWLIB_LIBC AND
     NOT (CONFIG_PICOLIBC AND NOT CONFIG_PICOLIBC_USE_MODULE) AND
     NOT COMPILER STREQUAL "xcc" AND
     NOT CONFIG_HAS_ESPRESSIF_HAL AND
+    NOT CONFIG_LLVM_LIBCXX AND
```

Comment:
Actually, it turned out that it's not necessary. Under this 'if' statement we set `-nostdinc` compiler flag, which has nothing to do with C++ (but it wasn't breaking anything because libc++ depends on full libc - newlib or picolibc). I think that I commited that accidentally and forgot to remove. Sorry!